WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/CardPresentPaymentsOnboardingUseCase.swift

```diff
@@ -48,15 +49,22 @@ final class CardPresentPaymentsOnboardingUseCase: CardPresentPaymentsOnboardingU
     init(
         storageManager: StorageManagerType = ServiceLocator.storageManager,
         stores: StoresManager = ServiceLocator.stores,
-        featureFlagService: FeatureFlagService = ServiceLocator.featureFlagService
+        featureFlagService: FeatureFlagService = ServiceLocator.featureFlagService,
+        cardPresentPaymentOnboardingStateCache: CardPresentPaymentOnboardingStateCache = CardPresentPaymentOnboardingStateCache.shared
     ) {
         self.storageManager = storageManager
         self.stores = stores
         self.configurationLoader = .init(stores: stores)
         self.cardPresentPluginsDataProvider = .init(storageManager: storageManager, stores: stores, configuration: configurationLoader.configuration)
         self.featureFlagService = featureFlagService
+        self.cardPresentPaymentOnboardingStateCache = cardPresentPaymentOnboardingStateCache
 
-        updateState()
+        // Rely on cached value if there's any
+        if let cachedValue = cardPresentPaymentOnboardingStateCache.value {
+            state = cachedValue
+        } else {
+            updateState()
```

Comment:
Hey, @toupper! This PR is super old but maybe you have a crazily good memory or understanding of this code 😀

In other places, `CardPresentPaymentOnboardingStateCache.shared.update(state)` is called after updating the state. But it's not called here, do you know if there's a reason for that? 

`POSEligibilityChecker` relies on this use case state. However, when the menu is opened, there's a slight delay in getting the state even though it was loaded when the use case was initialized. It happens because there's no caching here and it gets fetched again.

Reply:
Hi @staskus! 

It's been a while indeed, but I think I got the rationale behind that:

At the moment of the PR creation, there was no need to update the cache, as we already had the value on the state property. The cached value is retrieved right on the app start thanks to the `CardPresentPaymentsOnboardingIPPUsersRefresher`, so in theory, you shouldn't need to update the cache again. But if you need it now, I think it won't hurt to update the cache with the latest value you retrieve there :)